src/transport.js

```diff
@@ -203,23 +203,61 @@ class TransportManager {
     if (!transport || !transport.filter) return []
 
     const transportAddrs = transport.filter(multiaddrs)
-    if (!peerInfo) {
+    if (!peerInfo || !transportAddrs.length) {
       return transportAddrs
     }
 
-    const ourAddrs = peerInfo.multiaddrs.toArray()
-    return transportAddrs.filter((addr) => {
+    const ourAddrs = ourAddresses(peerInfo)
```

Comment:
I'm wondering if we should create and store this after listening is done, instead of running it before every dial. Our addresses shouldn't change, at the moment, after listening is done. It might save us quite a bit of processing when there are a lot of dials being done.

Reply:
I guess we could. It sounds like it would add a bunch of complexity that might easily cause confusion/bugs in the in the future if/when our addresses can change. Can we do it in a separate PR if it's a bottleneck?

Really, a lot of this should go away when addrs don't automatically carry a peer ID.

Reply:
I am ok with addressing this in a new PR

Reply:
> Really, a lot of this should go away when addrs don't automatically carry a peer ID.

That's true, and since it's on my list of things to get to this quarter I think we can move forward with this. The queuing should mitigate the performance impact anyway.